pkg/agent/client/client.go

```diff
@@ -171,8 +171,16 @@ func (c *client) RenewSVID(ctx context.Context, csr []byte) (*X509SVID, error) {
 	})
 	if err != nil {
 		c.release(connection)
+		errToReturn := fmt.Errorf("failed to renew agent: %w", err)
+		if s, ok := status.FromError(err); ok {
+			c.c.Log.WithFields(logrus.Fields{
+				"code":    s.Code(),
+				"message": s.Message(),
+			}).WithError(err).Error("Failed to renew agent")
+			return nil, errToReturn
+		}
 		c.c.Log.WithError(err).Error("Failed to renew agent")
```

Comment:
Thanks @rturner3! Fixed on [1bba9e2](https://github.com/spiffe/spire/pull/4262/commits/1bba9e28a704cfabfb86fe8134974bff79bbc10b)
I was considering creating constants for the strings `code` and `message` somewhere on the `names.go` file, which is in the `telemetry` package. Do you think it would be useful? I do, because we could reference them when handling the errors of others gRPC calls.

Also, I noticed that the following block is being repeated. Is there a package where I could create a function that set those fields to the logger? (if you think it would be useful).
```go
if s, ok := status.FromError(err); ok {
	log = log.WithFields(logrus.Fields{
		"code":    s.Code(),
		"message": s.Message(),
	})
}
```

Reply:
Sorry, I missed this comment before, but I agree that this is a good suggestion and think your approach for this in the PR looks good.